components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/AuthenticationHandler.java

```diff
@@ -124,8 +126,9 @@ protected void postAuthenticate(MessageContext messageContext, AuthenticationRes
                         authenticatedUser = (AuthenticatedUser) user;
                         // For B2B organization users, set the user ID which is set as username in user object.
                         if (authenticatedUser.isFederatedUser() && StringUtils.isNotEmpty(authorizedOrganization)) {
```

Comment:
As for now, federated user from enterprise IDPs can't switch to another organization (authorizedOrganization == null)